dynamodb/src/main/scala/uk/gov/nationalarchives/DADynamoDBClient.scala

```diff
@@ -31,6 +31,30 @@ class DADynamoDBClient[F[_]: Async](dynamoDBClient: DynamoDbAsyncClient):
   extension [T](completableFuture: CompletableFuture[T])
     private def liftF: F[T] = Async[F].fromCompletableFuture(Async[F].pure(completableFuture))
 
+  /** Deletes a list of items with primary keys of type T from Dynamo. Unprocessed items will be retried
+    *
+    * @param tableName
+    *   The name of the table
+    * @param primaryKeys
+    *   A list of primary keys to delete from Dynamo (list can be any length; requests will be batched into groups of
+    *   25)
+    * @param format
+    *   An implicit scanamo formatter for type T
+    * @tparam T
+    *   The type of the primary keys to be deleted from Dynamo
+    * @return
+    *   The List of BatchWriteItemResponse wrapped with F[_]
+    */
+  def deleteItems[T](tableName: String, primaryKeys: List[T])(using DynamoFormat[T]): F[List[BatchWriteItemResponse]] =
```

Comment:
1. Should we call this method `deleteItemsWithKey` or `deleteItemsByKey`?
2. Should we limit the upper bound of T to `Product` ( [T <: Product]) like `writeItems`?

Reply:
1. There's only one way to delete items which is by key so this feels a tad superfluous
2. There's another PR to get rid of all the `Product` among other things. As I say above, it's annoying if you want to pass traits which is perfectly valid.